GameOfRunes/GameOfRunes/GameEngine/ContactDelegate.swift

```diff
@@ -35,18 +35,31 @@ class ContactDelegate: NSObject, SKPhysicsContactDelegate {
     }
     
     private func enemyNodeContactWithOther(enemyNode: SKNode, other: SKNode) {
-        guard let enemyEntity = enemyNode.entity as? EnemyEntity else {
-            return
+        guard let enemyEntity = enemyNode.entity as? Entity,
+            enemyEntity.type == .enemyEntity,
+            let otherEntity = other.entity as? Entity else {
+                return
         }
-       
-        if let powerUpEntity = other.entity as? PowerUpEntity {
-            didActivate(powerUp: powerUpEntity.powerUpType, on: enemyEntity)
-        } else if other.entity is EndPointEntity {
+        
+        switch otherEntity.type {
+        case .endPointEntity:
             gameEngine?.enemyReachedLine(enemyEntity)
+        case .hellFirePowerUpEntity, .icePrisonPowerUpEntity, .darkVortexPowerUpEntity:
```

Comment:
`.darkVortex...` doesn't affect anything here right?

Reply:
yeah true, was trying to think of a way to refactor the powerups into a single type, but can't think of any